### PR TITLE
Remove RegisterITF references from OMROptions

### LINKED ISSUE #3961: Remove RegisterITF references from OMROptions
Register interferences are deprecated and largely removed from the codebase.  Remove the remaining remnants by grepping for `RegisterITF` and cleaning up the references.

You should only find references in OMROptions.cpp/.hpp.

### PR DESCRIPTION
**Pull request by nidhi1605**
Fixes #3961
Signed-off-by: Nidhi Gupta <itsnidhi16@gmail.com>

### CODE REVIEW

File: compiler/control/OMROptions.hpp [hunk 1]
```diff
@@ -1050,14 +1050,6 @@ enum TR_CompilationOptions
    TR_TraceLRAResults                   = 0x00000800,
    // Available                         = 0x00001000,
 
-   // Register ITF tracing option word
-   //
-   TR_TraceRegisterITFBasic             = TR_TraceGRABasic,
-   // Avaialble                         = 0x00008000,
-   TR_TraceRegisterITFBuild             = 0x00010000,
-   // Available                         = 0x00020000,
-   TR_TraceRegisterITFColour            = 0x00040000,
```
Comment (Leonardo2718) [inline compiler/control/OMROptions.hpp]: The first of these fields (`TR_TraceRegisterITFBasic`) can be safely removed. However, subsequent ones should stay and be commented out as `// Available`. This allows us to keep track of which bits are available for future options.

```c++
   // Avaialble                         = 0x00008000,
   // Avaialble                         = 0x00010000,
   // Available                         = 0x00020000,
   // Avaialble                         = 0x00040000,
```

File: compiler/control/OMROptions.hpp [hunk 2]
```diff
@@ -1049,7 +1049,14 @@ enum TR_CompilationOptions
    //
    TR_TraceLRAResults                   = 0x00000800,
    // Available                         = 0x00001000,
-
+  
+   // Register ITF tracing option word
```
Comment (Leonardo2718) [inline compiler/control/OMROptions.hpp]: This comment is no longer relevant.

**Comment by Leonardo2718** [comment]
@nidhi1605 can you please squash the commits together?

**Comment by nidhi1605** [comment]
I have squashed the commits together, @Leonardo2718 could you please review this?

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@Leonardo2718 , could you sign off on these changes please?

**Pull request merged**